GitUI/UserControls/RevisionGrid/Graph/RevisionGraphRow.cs

```diff
@@ -102,7 +103,7 @@ private void BuildSegmentLanes()
                         // Segment has not been assigned a lane yet
                         if (!added)
                         {
-                            newSegmentLanes.Add(segment, laneIndex);
+                            newSegmentLanes[segment] = laneIndex;
```

Comment:
This has hit me a couple of times - fixed before I could report or investigate!

Reply:
This is not the cause of the problem. I fixed the cause in this PR. After the other changes, I never got an argument exception anymore. But this is ok. There are other guards that validate if something goes wrong.